lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,451 @@
+
+import logging
+
+from django.core.urlresolvers import reverse
+from django.http import Http404
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework import generics
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+
+from courseware.courses import course_image_url
+from course_api.courseware_access import get_course, get_course_child, get_course_descriptor
+from openedx.core.lib.api.views import APIViewWithPermissions, PaginatedListAPIViewWithPermissions, PermissionMixin
+from course_api.v0.serializers import CourseSerializer, AssignmentSerializer, GradingPolicySerializer
+
+
+log = logging.getLogger(__name__)
+
+
+def _get_content_children(content):
+    """
+    Parses the provided content object looking for children.
+    """
+    children = []
+    if content.has_children:
+        child_content = content.get_children()
+        for child in child_content:
+            children.append(child)
+    return children
+
+
+def _serialize_content(request, course_key, content_descriptor):
+    """
+    Loads the specified content object into the response dict
+    This should probably evolve to use DRF serializers
+    """
+
+    data = {}
+
+    if hasattr(content_descriptor, 'display_name'):
+        data['name'] = content_descriptor.display_name
+
+    if hasattr(content_descriptor, 'due'):
+        data['due'] = content_descriptor.due
+
+    data['start'] = getattr(content_descriptor, 'start', None)
+    data['end'] = getattr(content_descriptor, 'end', None)
+
+    data['category'] = content_descriptor.location.category
+
+    # Some things we only do if the content object is a course
+    if hasattr(content_descriptor, 'category') and content_descriptor.category == 'course':
+        content_id = unicode(content_descriptor.id)
+        content_uri = request.build_absolute_uri(reverse('course_api:v0:detail', kwargs={'course_id': content_id}))
+        data['course'] = content_descriptor.location.course
+        data['org'] = content_descriptor.location.org
+        data['run'] = content_descriptor.location.run
+
+    # Other things we do only if the content object is not a course
+    else:
+        content_id = unicode(content_descriptor.location)
+        # Need to use the CourseKey here, which will possibly result in a different (but valid)
+        # URI due to the change in key formats during the "opaque keys" transition
+        content_uri = request.build_absolute_uri(reverse('course_api:v0:content:detail',
+                                                         kwargs={'course_id': unicode(course_key),
+                                                                 'content_id': content_id}))
+
+    data['id'] = unicode(content_id)
+    data['uri'] = content_uri
+
+    # Include any additional fields requested by the caller
+    include_fields = request.QUERY_PARAMS.get('include_fields', None)
```

Comment:
`format` is the XBlock field that groups particular types of grades together (it associates content with a grading policy). So, in the average course, querying by a particular format type will let you distinguish homeworks and exams, but I worry if you're going to hard-code particular query values on the analytics end of things (rather than just selecting from the set of formats defined in the course grading policy).